### ISSUE TITLE
Stuck Connecting to Network

### ISSUE THREAD

**Issue by cardinalproperty** [input-output-hk/daedalus#508]
[LOG.txt](https://github.com/input-output-hk/daedalus/files/1346009/LOG.txt)
I'm running Windows 10 and just installed the Daedalus wallet.  It just hangs saying it's "Connecting to network..." I've reopened the app, and tried View > Reload, but it still just says it's connecting.

By connecting, does it mean it's downloading a blockchain or something?  I'm accustomed to that taking a while with other wallets, but given that mainnet just launched I didn't think that would be it.

Let me know if I'm just being impatient (I gave it about 90 minutes the last time I launched it).

**Comment by darko-mijic** [comment]
cc @fersel

**Comment by fersel** [comment]
@cardinalproperty please take a look at our FAQ section at https://daedaluswallet.io/faq/ and see if that helps. If not, we'll need your node's logs (details on how to obtain them are provided on the same page).

**Comment by cardinalproperty** [comment]
@fersel thanks for the response.  I read the FAQ section on what to do if you're stuck on that screen, but no dice.
I made sure nothing relevant was blocked by Windows Firewall, launched through the desktop shortcut and then through the .bat file, but still hung on connecting screen for several minutes.

I've attached the whole logs folder as a .zip.
[Logs.zip](https://github.com/input-output-hk/daedalus/files/1346456/Logs.zip)

**Comment by cardinalproperty** [comment]
Bumping.  I completely uninstalled and reinstalled the Daedalus wallet today, and this is still happening.  Any insights on why?

**Comment by pawodrep** [comment]
I had similar problem. Going through logs, in _cardano-node.log_ file, I spotted line:
`cardano-node.exe: user error (open: IO error: Failed to create dir: cardano-node.exe: user error (open: IO error: Failed to create dir: C:\Users\Pawe�\AppData\Roaming\Daedalus\DB-1.0\blocks\index: No such file or directory)`
I've tried running the wallet with administrator rights but it did not work. After looking for similar problems online and I found out that this may be due to non-ascii characters in path (user folder name). After renaming user folder the application started to work normally (I've got connected, syncing started, etc.). So I guess that this could be related to improper handling of unicode characters.

**Comment by hydro13** [comment]
I have helpt some users by : stop program. Rename or delete C:\Users\"user-name"\AppData\Roaming\Daedalus
Reinstall from https://daedaluswallet.io/#download
Import your wallet 
and its working 👍  the problem must be somwhere in AppData\Roaming\Daedalus

**Comment by cardinalproperty** [comment]
Thanks for the help, @hydro13; still no luck, but I appreciate the effort.

I tried reinstalling, and it's doing the connecting to network... thing now, but I'll let it run for a bit to see if a miracle happens, since the version number is higher than the last time I installed it.  I bought some ADA as soon as it became available on bittrex, but it's still sitting on the exchange.  I'll have to just sell it if I can't get the wallet to work.

**Comment by hydro13** [comment]
Your on a slow computer? edit your Daedalus.bat  change  --nodetimeout 5 to --nodetimeout 30 save file.
you need to edit the file in notepad as administrator (run as administrator)..
start the wallet with the bat-file

**Comment by domenkozar** [comment]
@cardinalproperty do you still have the issue?

**Comment by Jawabiscuit** [comment]
Hi, this is still an issue.

**Comment by lowtempo711** [comment]
No more issue.  Thanks!

**Comment by bitcartel** [comment]
Changing the timeout fixes the problem

**Comment by csoroz** [comment]
Truing changing --node-timeout from 5 TO 30 as said above... still not working.

**Comment by csoroz** [comment]
The issue persist with the new version Daedalus 1.0.3769

**Comment by humanoidcreature** [comment]
I have this also. Pretty frustrating.

**Comment by cleverca22** [comment]
@humanoidcreature are you on windows?, what the absolute path to your desktop? like `C:\users\<username>\desktop` ?

**Comment by humanoidcreature** [comment]
@cleverca22 I'm using Windows 10 and yes the path is correct. (I've got accent in my name, but it would be crazy if that's the problem.)

**Comment by cleverca22** [comment]
yep, there is a bug with unicode characters in your username

**Comment by csoroz** [comment]
I'm also using Windows 10 and my <username> is four ASCII chars, but the issue still persist.

**Comment by cleverca22** [comment]
@csoroz are there any obvious errors in `C:\Users\USER\AppData\Roaming\Daedalus\Logs\cardano-node.log`

**Comment by csoroz** [comment]
@cleverca22 Yes there are Timeout errors. I send you the [Logs.zip](https://github.com/input-output-hk/daedalus/files/1548452/Logs.zip)

In `daedalus.bat`, variable `%DAEDALUS_DIR%` has an `\` at the end, so some paths have a duplicated `\`. I have tried correcting that in the .bat file, but does not solve the problem.

**Comment by cleverca22** [comment]
@csoroz does your firewall/isp allow tcp port 3000 outgoing?

**Comment by csoroz** [comment]
@cleverca22 I have tried allowing port 3000 in windows firewall and also opening it in router configuration. No success...

Making `ping` to those ip addresses fails, for example:

    C:>ping 13.115.112.96
    Pinging 13.115.112.96 with 32 bytes of data:
    Request timed out.

http://downforeveryoneorjustme.com/13.115.81.99

**Comment by cleverca22** [comment]
@csoroz normal ping is blocked, only tcp port 3000 is open, try something more like telnet and see if it can even connect

**Comment by csoroz** [comment]
Telnet to port 3000 seems to work but I only get a blank screen then finalizes...

**Comment by csoroz** [comment]
@cleverca22 Testing with http://portquiz.net:3000
I can reach port 3000.
Daedalus still fails to connect.

**Comment by lucianboboc** [comment]
I have the same problem on macOS Sierra with the last wallet version.
Also 9/10 times when i try to open the app it just jumps in the launch bar and then it remains there freezing "Application not responding".
I have this 2 problems since i first installed this app, what is going on?

**Comment by mimim-bkk** [comment]
So far the only way I found is to connect my macbook to 4G with my iPhone using personal hotspot and Daedalus can connect to the network and start to sync. 
So my guess would be a router setup issue, I tried many combinations on port forwarding nor different network configurations but no success.
If someone have an idea how to setup the router to make it work, please share.

**Comment by gufmar** [comment]
As it looks like a noticeable number of users has problems using Daedalus just because their computer (AV Suite) or local network (LAN/WAN) does not allow connections to the 7 bootstrap servers listening on port 3000 I would suggest introducing a **fallback on port 443** (HTTPS) or a complete switch to such a **standard port**.

**Comment by gufmar** [comment]
Also Windows 8/10 users should be averted that the current launch method (daedalus.bat in Start menu and on desktop) should be used to start the cardano-node and the daedalus wallet.

If after a first start the user right click the daedalus icon in the task bar to pin it there, this does only pin daedalus.exe. So on next start the wallet will stuck on "connecting to network..." because there is no cardone-node up & running.

**Comment by gufmar** [comment]
Last but not least both the possible issue with port 3000 and the task bar pinning should be mentionedn on the public FAQ page.

**Comment by csoroz** [comment]
For the first time I've seen Daedalus pass the "Connecting to network..." screen. I haven't done anything special (maybe some Windows 10 update?). I let it for a while syncing to the network... when done, I've closed the program. Subsequent runs have failed again.
Now it has succeeded for the second time...

**Comment by gufmar** [comment]
Note: I placed the request to use standard (well known) ports also on the devOps repository.
After I slept on it and reading about Daedalus philosophy (general crypto client, app store, ...) I believe the same thing should be considered for Daedalus current port 8090. When it becomes a light-weight client running on all kind of devices and in all kind of IP-networks, in order to connect to cardano-nodes using standard ports like 443 (usually used but not exclusively limited to https) wouldn't have any drawbacks. But it could avoid this low but definitively existing bunch of situations, where this connection attempts are blocked by some security/firewall rules. 
So this is not rocket science or a special thing. But I would say it pairs very good with all your efforts to provide high assurance solutions.

**Comment by snajpa** [comment]
I have hit this this twice today on two different Windows 10 PCs. Daedalus log says ECONNREFUSED, I have yet to investigate why; but: those two machines are running localized Czech version of Windows, although they don’t use unicode chars in usernames. It syncs once, I can restore the wallet, but I can’t open Daedalus again after closing it (no, theres nothing running on :8090).

**Comment by csoroz** [comment]
It's working for me since my previous comment...

**Comment by 2072** [comment]
I have the same problem on Windows 10 (using official build 1.0.3769). Tried all solutions mentioned in the FAQ to no avail.
There is no obvious error in the logs apart from a few "Reporting error" errors:

```log
[node:INFO:ThreadId 3] [2018-01-21 19:21:55 UTC] Our verInfo: VerInfo { magic=2d964a09, blockVersion=0.1.0, inSpecs={96: Conv 43, 49: Conv 5c, 97: Conv 3d, 34: Conv 5e, 98: Conv 37, 67: Conv 60, 83: Conv 0, 4: Conv 5, 5: Conv 4, 37: Conv 5e, 6: Conv 7, 55: Conv 62, 73: Conv 5f, 43: Conv 5d, 92: Conv 31, 61: Conv 61, 93: Conv 2b, 94: Conv 25, 95: Conv 49}, outSpecs={49: Conv 5c, 67: Conv 60, 83: Conv 0, 4: Conv 5, 5: Conv 4, 37: Conv 5e, 6: Conv 7, 55: Conv 62, 73: Conv 5f, 43: Conv 5d, 13: Conv 0, 61: Conv 61}}
[node:DEBUG:ThreadId 3] [2018-01-21 19:21:55 UTC] startNode, we are NodeId 86521250-6f37-44dd-be9f-47c42600f006
[node:INFO:ThreadId 3] [2018-01-21 19:21:55 UTC] cardano-sl: commit 0c1fab91a7358fcb4c151a3a028ed650f7a36cbd
[node:INFO:ThreadId 3] [2018-01-21 19:21:55 UTC] Application: csl-daedalus:3, last known block version 0.1.0
[node:INFO:ThreadId 3] [2018-01-21 19:21:55 UTC] Genesis stakeholders (7 addresses, dust threshold 7 coin(s)): GenesisWStakeholders: {1deb8290: 1, 43011479: 1, 5071d880: 1, 5411c7bf: 1, 65904a89: 1, 6c9e1497: 1, af2800c1: 1}
[node:INFO:ThreadId 3] [2018-01-21 19:21:55 UTC] First genesis block hash: 89d9b5a5, genesis seed is 76617361206f7061736120736b6f766f726f64612047677572646120626f726f64612070726f766f6461
[node:INFO:ThreadId 3] [2018-01-21 19:21:55 UTC] Last known leaders for epoch epoch #0 are: [STRIPPED from the logs because it's really too long]
[node:INFO:ThreadId 3] [2018-01-21 19:21:55 UTC] Current tip header: GenesisBlockHeader:
    hash: 89d9b5a5b8ddc8d7e5a6795e9774d97faf1efea59b2caf7eaf9f8c5b32059df4
    previous block: 5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb
    epoch: epoch #0
    difficulty: 0

[node.worker.subscription:NOTICE:ThreadId 561] [2018-01-21 19:21:55 UTC] dnsSubscriptionWorker: valency 1
[node.plugin:INFO:ThreadId 562] [2018-01-21 19:21:55 UTC] DAEDALUS has STARTED!
[node:ERROR:ThreadId 561] [2018-01-21 19:21:55 UTC] Reporting error with reason "Worker/plugin with logger name LoggerName {loggerName = "node"} failed with exception: ParseError {errorContexts = [], errorMessage = "not enough input", errorPosition = 1:1}"
[node.wallet.servant:INFO:ThreadId 579] [2018-01-21 19:21:55 UTC] 
GET
    :> api
    :> settings
    :> sync
    :> progress
   Status: OK 0s
[node.wallet.servant:INFO:ThreadId 588] [2018-01-21 19:21:55 UTC] 
GET
    :> api
    :> settings
    :> sync
    :> progress
   Status: OK 0s
```
the last log entry is then followed by countless similar `node.wallet.servant:INFO` entries with the exact same data apart from times and `ThreadId`

and occasionally by:

```log
[node:ERROR:ThreadId 561] [2018-01-21 19:22:45 UTC] Didn't manage to report RError "Worker/plugin with logger name LoggerName {loggerName = \"node\"} failed with exception: ParseError {errorContexts = [], errorMessage = \"not enough input\", errorPosition = 1:1}" because of exception 'Failed to send a report, the exception was: HttpExceptionRequest Request {
  host                 = "report-server.cardano-mainnet.iohk.io"
  port                 = 8080
  secure               = False
  requestHeaders       = [("Content-Type","multipart/form-data; boundary=----WebKitFormBoundaryUmhUaEIpojUScQHk")]
  path                 = "/report"
  queryString          = ""
  method               = "POST"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
 (ResponseBodyTooShort 218949 141)' raised while sending
```

Monitoring the process using process explorer shows that apart from listening locally on port 8090 and connecting to `ec2-18-194-141-68.eu-central-1.compute.amazonaws.com` on port `8080` at startup, cardano-node.exe is not attempting any connection to the outside.

Using process monitor, I can see it tries to open these non existant registry keys in loop:

```log
HKLM\Software\Microsoft\Cryptography\Offload
HKLM\Software\Microsoft\Cryptography\DESHashSessionKeyBackward
HKLM\Software\Microsoft\Cryptography\Offload
HKLM\Software\Microsoft\Cryptography\DESHashSessionKeyBackward
```
At startup it also tries to access hundreds of non existant registry keys under `HKLM/System`

I hope this helps to track down the problem.

**Comment by 2072** [comment]
Ok commit 3f7e0338b6a89172380b685a6c8514be30b15b39 put me on the track. it was a DNS issue for me:   ```relays.cardano-mainnet.iohk.io``` does  not resolve using my configured DNS. So like in the aforementioned commit I put addresses directly (the same as in the actual DNS record) and now it's syncing...

**Comment by kk88london** [comment]
You guys are all super smart, I am not a techie, but I managed to fix my issue, after trying all the above. Not sure which change fixed it, but most likely switching from my home wifi to my mobile hotspot. It's now syncing. Let's see how that goes.

**Comment by mykitou** [comment]
Guys and girls, I just made my inscription because I think I found the problem. Create a new session in windows with admin rights, and in the name of the session do but input spaces or accents as é. Put a session name like user1 or admin for exemple.
Reinstall the wallet on the new session and executive the bat in the installation folder and everything right. 
Sorry for my English I'm french.

Hop will help some of you

If you are happy make me happy

Ada


DdzFFzCqrhsyupVRGetsCTGkkZPrrotTmGaYrVipGjF4jf8DWQto5QSY6y7D3w7kUqzxBF57RDR5GK6yfbUtCVttATvtkdk8i8FH7e1m
